trl/trainer/bco_config.py

```diff
@@ -202,3 +202,9 @@ class BCOConfig(TrainingArguments):
         default=10.0,
         metadata={"help": "Maximum value of the density ratio. The estimated density ratio is clamped to this value."},
     )
+
+    def __post_init__(self):
+        if self.bf16 is None:
+            self.bf16 = True if not self.fp16 else False
+
```

Comment:
when `bf16` is set, directly pass to `TrainingArguments` `post_init_` to do argument validation and setting.
when `bf16` is not set, if `fp16` is `True`, set `bf16` to `False`, else to `True`